src/Catalyst.Core.Modules.Ledger/Web3EthApi.cs

```diff
@@ -22,25 +22,76 @@
 #endregion
 
 using System;
+using Catalyst.Abstractions.Consensus.Deltas;
+using Catalyst.Abstractions.Hashing;
+using Catalyst.Abstractions.IO.Events;
 using Catalyst.Abstractions.Kvm;
 using Catalyst.Abstractions.Ledger;
+using Catalyst.Abstractions.Ledger.Models;
+using Catalyst.Abstractions.P2P;
+using Catalyst.Abstractions.Repository;
+using Catalyst.Core.Lib.Extensions;
+using Catalyst.Core.Modules.Ledger.Repository;
+using Catalyst.Protocol.Deltas;
+using Catalyst.Protocol.Peer;
+using Catalyst.Protocol.Transaction;
+using Catalyst.Protocol.Wire;
+using Lib.P2P;
+using Nethermind.Core.Crypto;
 using Nethermind.Store;
 
-namespace Catalyst.Core.Modules.Ledger 
+namespace Catalyst.Core.Modules.Ledger
 {
     public class Web3EthApi : IWeb3EthApi
     {
-        public Web3EthApi(IDeltaExecutor deltaExecutor, IStateReader stateReader, IStateRootResolver stateRootResolver, IDeltaResolver deltaResolver)
+        private readonly ITransactionRepository _receipts;
+        private readonly ITransactionReceivedEvent _transactionReceived;
+        public IHashProvider HashProvider { get; }
+        private readonly PeerId _peerId;
+
+        public Web3EthApi(IStateReader stateReader, IDeltaResolver deltaResolver, IDeltaCache deltaCache, IDeltaExecutor executor, IStorageProvider storageProvider, IStateProvider stateProvider, ITransactionRepository receipts, ITransactionReceivedEvent transactionReceived, IHashProvider hashProvider, IPeerSettings peerSettings)
         {
-            DeltaExecutor = deltaExecutor ?? throw new ArgumentNullException(nameof(deltaExecutor));
+            _receipts = receipts;
+            _transactionReceived = transactionReceived ?? throw new ArgumentNullException(nameof(transactionReceived));
+            HashProvider = hashProvider;
+            _peerId = peerSettings.PeerId;
+
             StateReader = stateReader ?? throw new ArgumentNullException(nameof(stateReader));
-            StateRootResolver = stateRootResolver ?? throw new ArgumentNullException(nameof(stateRootResolver));
             DeltaResolver = deltaResolver ?? throw new ArgumentNullException(nameof(deltaResolver));
+            DeltaCache = deltaCache ?? throw new ArgumentNullException(nameof(deltaCache));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The parameter name '' is not declared in the argument list.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4979613)